packages/node/src/utils.ts

```diff
@@ -0,0 +1,38 @@
+import * as fs from 'fs';
+import * as path from 'path';
+
+/**
+ * Recursively read the contents of a directory.
+ *
+ * @param targetDir Absolute or relative path of the directory to scan. All returned paths will be relative to this
+ * directory.
+ * @returns Array holding all relative paths
+ */
+export function deepReadDirSync(targetDir: string): string[] {
+  const targetDirAbsPath = path.resolve(targetDir);
+
+  if (!fs.existsSync(targetDirAbsPath)) {
+    throw new Error(`Cannot read contents of ${targetDirAbsPath}. Directory does not exist.`);
+  }
+
+  if (!fs.statSync(targetDirAbsPath).isDirectory()) {
+    throw new Error(`Cannot read contents of ${targetDirAbsPath}, because it is not a directory.`);
+  }
+
+  // This does the same thing as its containing function, `deepReadDirSync` (except that - purely for convenience - it
+  // deals in absolute paths rather than relative ones). We need this to be separate from the outer function to preserve
+  // the difference between `targetDirAbsPath` and `currentDirAbsPath`.
+  const deepReadCurrentDir = (currentDirAbsPath: string): string[] => {
```

Comment:
Any reason for this to be a nested function?

Reply:
It's a recursive function that is not used outside of the parent function, so it's here for the convinience.

Reply:
> ```
> // We need this to be separate from the outer function to preserve
> // the difference between `targetDirAbsPath` and `currentDirAbsPath`.
> ```

Or do you mean why did I define it inside of `deepReadDirSync` instead of outside of it? Just for organization purposes. If that is what you mean, I'm curious - is there a reason you think it shouldn't be inside?